sdk/internal/recording/recording.go

```diff
@@ -624,45 +659,34 @@ func GetRecordingId(t *testing.T) string {
 }
 
 func GetRecordMode() string {
+	if !(recordMode == RecordingMode || recordMode == PlaybackMode || recordMode == LiveMode) {
+		recordMode = PlaybackMode
+	}
 	return recordMode
 }
 
-func getRootCas(t *testing.T) (*x509.CertPool, error) {
-	localFile, ok := os.LookupEnv("PROXY_CERT")
-
-	rootCAs, err := x509.SystemCertPool()
-	if err != nil && strings.Contains(err.Error(), "system root pool is not available on Windows") {
-		rootCAs = x509.NewCertPool()
-	} else if err != nil {
-		return rootCAs, err
+func findProxyCertLocation() (string, error) {
+	fileLocation, ok := os.LookupEnv("PROXY_CERT")
+	if ok {
+		return fileLocation, nil
 	}
 
+	// Couldn't be found, lets try to find it from the GOPATH location
+	var err error
+	goPath, ok := os.LookupEnv("GOPATH")
 	if !ok {
-		t.Log("Could not find path to proxy certificate, set the environment variable 'PROXY_CERT' to the location of your certificate")
-		return rootCAs, nil
-	}
-
-	cert, err := ioutil.ReadFile(localFile)
-	if err != nil {
-
-		return nil, err
-	}
-
-	if ok := rootCAs.AppendCertsFromPEM(cert); !ok {
-		t.Log("No certs appended, using system certs only")
+		goPath, err = os.UserHomeDir()
+		if err != nil {
+			return "", fmt.Errorf("could not find the home directory using `os.UserHomeDir()`, %s", err.Error())
+		}
 	}
 
-	return rootCAs, nil
+	goPath = filepath.Join(goPath, "src", "github.com", "Azure", "azure-sdk-for-go", "eng", "common", "testproxy", "dotnet-devcert.crt")
```

Comment:
Forgive the newbie gopher question but would this path typically exist in the workspace of an SDK developer? It doesn't exist in mine, and now I wonder whether I've taken an unconventional approach in setting up my workspace

Reply:
I don't think we can make assumptions about the path, and given that GOPATH is going away, shouldn't rely on it either.

Reply:
When is GOPATH slated to be removed @jhendrixMSFT ?

Reply:
I don't know when, and I suppose it's likely to be here from some time.

With the introduction of modules, there's no longer a need to explicitly set it or place one's source code inside it.  For example, my GOPATH from `go env`.

```
set GOPATH=C:\Users\jhendrix\go
```

I don't place any of my source code in this directory, it's all under various subdirectories in `c:\git`.

Reply:
That's unfortunate for this scenario. I think the best case is to now remove all this and only search in the environment variable, agreed @jhendrixMSFT ?

Reply:
If this is needed to find files within the tree you could execute this to get the repo root.
```
git rev-parse --show-toplevel
```

Returns something like `C:/git/Azure/go-amqp` or whatever.

I'll also say this is why having test assets under `<package>/testdata` makes sense as you can just use a relative path (I get this scenario is a bit odd).

Reply:
I'll try using the git command to do so, if that doesn't work well I'll remove this bit of code. We would have to copy the cert to every package/testdata folder which is more work than asking everyone to set the environment variable